### PR TITLE
Clean up error message when Razor compiler is not referenced

### LINKED ISSUE #1974: [Razor SDK]: Fit and polish - Determine what the right way is to indicate an error when Razor.Design is missing
Based on https://github.com/aspnet/Razor/pull/1964/files#r163731196

The Razor.SDK requires CodeGeneration targets to be present to work. Need to ask around with folks more familiar with writing SDKs what is the quintessential MSBuild approach for this is.

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #1974

### CODE REVIEW

File: src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets [hunk 1]
```diff
@@ -482,8 +482,11 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   </Target>
 
-  <Target Name="_EnsureRazorCompilerReferenced" Condition="'$(IsRazorCompilerReferenced)' != 'true'">
-    <Error Text="A reference to Microsoft.AspNetCore.Razor.Design is required for this target to succeed. See todo:fwlink for instructions on how to resolve this." />
+  <Target Name="_CheckForMissingRazorCompiler" Condition="'$(IsRazorCompilerReferenced)' != 'true'">
+    <Error
+      Text="A PackageReference for 'Microsoft.AspNetCore.Razor.Design' was not included in your project. This package is required to compile Razor files. Typically, a
```
Comment (pranavkm) [inline src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets]: Losely based on an Sdk message (which is in a resx string so I can't copy it)

File: src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets [hunk 2]
```diff
@@ -482,8 +482,11 @@ Copyright (c) .NET Foundation. All rights reserved.
...
+      transitive reference to 'Microsoft.AspNetCore.Razor.Design' and references required to compile Razor files are obtained by adding a PackageReference
+      for 'Microsoft.AspNetCore.Mvc' in your project. For more information, see https://go.microsoft.com/fwlink/?linkid=868374." />
```
Comment (Eilon) [inline src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets]: Who will make this FWLink point to something useful? Right now it points to https://docs.microsoft.com/en-us/aspnet/core/, which doesn't seem terribly useful.

Comment (pranavkm) [reply src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets]: I have this item - https://github.com/aspnet/Razor/issues/2119 - to track doc work which includes fixing up the link to the relevant doc page.

Comment (Eilon) [reply src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets]: Great!

**Pull request merged**